File: verify_ux.py
import asyncio
from playwright.async_api import async_playwright
import os

async def run():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page(viewport={'width': 1600, 'height': 1200})
        
        task_id = "818c5ce2-8d92-4398-9719-99f003a57f57"
        url = f"http://localhost:3011/tasks/{task_id}"
        print(f"Opening {url}")
        await page.goto(url)
        await page.wait_for_load_state("networkidle")

        # Switch to Graph tab
        print("Switching to Graph tab...")
        await page.click("button:has-text('神经图谱')")
        await page.wait_for_timeout(2000)
        await page.screenshot(path="step1_graph_init.png")

        # Find a Process node and click it
        # We'll try to find a text element inside the SVG
        print("Clicking a Process node...")
        # In mock or real data, we usually have some process name. 
        # Since I don't know the exact name, I'll look for any <g> that contains 'Process' label text
        process_node = page.locator("g:has(text='Process')").first
        if await process_node.count() > 0:
            await process_node.click()
            await page.wait_for_timeout(1000)
            await page.screenshot(path="step2_process_selected.png")
            print("Process selected.")
        else:
            print("No Process node found in graph!")
            await browser.close()
            return

        # Click a Step (should be a code node, e.g. 'Function' or 'Method')
        print("Clicking a code node (step)...")
        code_node = page.locator("g:has(text='Function'), g:has(text='Method'), g:has(text='Class')").first
        if await code_node.count() > 0:
            await code_node.click()
            await page.wait_for_timeout(1000)
            await page.screenshot(path="step3_code_selected.png")
            print("Code node selected. Checking sidebar for back button...")
            
            back_button = page.locator("button:has-text('返回：')")
            if await back_button.is_visible():
                print("Back button is VISIBLE.")
            else:
                print("Back button is MISSING!")
        else:
            print("No code node found in graph!")

        # Navigate to another code node to test stickiness
        print("Navigating to another code node...")
        other_code_node = page.locator("g:has(text='Function'), g:has(text='Method')").nth(1)
        if await other_code_node.count() > 0:
            await other_code_node.click()
            await page.wait_for_timeout(1000)
            await page.screenshot(path="step4_sticky_check.png")
            
            back_button = page.locator("button:has-text('返回：')")
            if await back_button.is_visible():
                print("Back button is STILL VISIBLE (Sticky!).")
            else:
                print("Back button GONE (Not sticky!).")
                
            # Test return
            print("Clicking back button...")
            await back_button.click()
            await page.wait_for_timeout(1000)
            await page.screenshot(path="step5_returned.png")
            print("Returned to process.")
        
        await browser.close()

asyncio.run(run())

File: frontend/verify_ux.js
/* eslint-disable @typescript-eslint/no-require-imports */
const { chromium } = require('playwright');

async function run() {
  const browser = await chromium.launch();
  const page = await browser.newPage({ viewport: { width: 1600, height: 1200 } });
  
  const taskId = "818c5ce2-8d92-4398-9719-99f003a57f57";
  const url = `http://localhost:3011/tasks/${taskId}`;
  console.log(`Opening ${url}`);
  await page.goto(url, { waitUntil: 'load', timeout: 60000 });
  console.log("Page loaded.");

  // Switch to Graph tab
  console.log("Switching to Graph tab...");
  await page.click("button:has-text('神经图谱')");
  await page.waitForTimeout(5000); 
  await page.screenshot({ path: "step1_graph_init.png" });

  // 1. Click a Process node
  console.log("Clicking a Process node...");
  const processNode = page.locator('svg >> text=Process').first();
  await processNode.click();
  await page.waitForTimeout(1500);
  await page.screenshot({ path: "step2_process_selected.png" });
  console.log("Process selected.");

  // 2. Click a code node (should have cursor-pointer)
  console.log("Clicking a code node (step)...");
  const codeNode = page.locator('g.cursor-pointer').first();
  if (await codeNode.count() > 0) {
    await codeNode.click();
    await page.waitForTimeout(1500);
    await page.screenshot({ path: "step3_code_selected.png" });
    console.log("Code node selected.");
    
    const backButton = page.locator("button:has-text('返回：')");
    if (await backButton.isVisible()) {
      const btnText = await backButton.innerText();
      console.log(`Back button is VISIBLE: ${btnText}`);
    } else {
      throw new Error("Back button is MISSING!");
    }

    // 3. Navigate to another code node to test stickiness
    console.log("Navigating to another code node to test stickiness...");
    const otherCodeNodes = await page.locator('g.cursor-pointer').all();
    if (otherCodeNodes.length > 1) {
      await otherCodeNodes[1].click();
      await page.waitForTimeout(1500);
      await page.screenshot({ path: "step4_sticky_check.png" });
      
      if (await backButton.isVisible()) {
         console.log("Back button is STILL VISIBLE (Sticky!).");
      } else {
         throw new Error("Back button GONE after navigation (Not sticky!).");
      }
      
      // 4. Test return
      console.log("Clicking back button to return...");
      await backButton.click();
      await page.waitForTimeout(1500);
      await page.screenshot({ path: "step5_returned.png" });
      console.log("Successfully returned to process view.");
    }
  } else {
    console.log("No code nodes found in the graph.");
  }
  
  await browser.close();
}

run().catch(e => { console.error(e); process.exit(1); });
